test/geopackage/WKB_Test.cpp

```diff
@@ -0,0 +1,262 @@
+        };
+
+        // MULTILINESTRING((10 10,20 20,10 40),(40 40,30 30,40 20,30 10))
+        this->wkb_bytes["multilinestring"] = {
+            0x01,
+            0x05, 0x00, 0x00, 0x00,
+            0x02, 0x00, 0x00, 0x00,
+            0x01,
+            0x02, 0x00, 0x00, 0x00,
+            0x03, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x34, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x34, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0x40,
+            0x01,
+            0x02, 0x00, 0x00, 0x00,
+            0x04, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3e, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3e, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x44, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x34, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3e, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x40
+        };
+
+        // POLYGON ((0 0, 5 0, 5 5, 0 5, 0 0), (2 2, 3 2, 3 3, 2 3, 2 2))
+        this->wkb_bytes["polygon_with_holes"] = {
+            0x01,
+            0x03, 0x00, 0x00, 0x00,
+            0x02, 0x00, 0x00, 0x00,
+            0x05, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x05, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x40, 
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40
+        };
+    }
+
+    void TearDown() override {}
+    
+    std::map<std::string, std::vector<uint8_t>> wkb_bytes;
+};
+
+TEST_F(WKB_Test, wkb_point_test) // also tests endianness
```

Comment:
Re endian-ness, do you have some literal values in the above fixture both ways around, with the indicator to switch between them? If so, that's worth stating explicitly. I can't quite parse floating point from literal bytes in my head 😃

Reply:
That was exactly what I meant. Now that you've said it, I can see that they're equivalent up to byte-reversal of each piece. I suggested renamings so that it's more self-evident

Reply:
Implemented in 5531a89 based on your suggestion below.